ui: Sortable sort per page

vladlos opened the pull request:
moved pagination logic to `Sortedtable` component for fixed an issue with sort settings when sorting is worked out for one page

Resolves: #46173

Release justification: bug fix

Release note (ui): this fixes a bug where sort columns were only being applied per-page instead of for the entire multi-page list of statements

vladlos committed:
ui: Sortable sort per page

moved pagination logic to `Sortedtable` component for fixed an issue with sort settings when sorting is worked out for one page

Resolves: #46173

Release justification: bug fix

Release note (ui): none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46416)
<!-- Reviewable:end -->

dhartunian reviewed (changes requested):
Regarding testing, there are no tests to account for the modified behavior. Can you add a test to the `SortedTable` component that shows the pagination being applied after the sort?

Additionally, this commit does not pass existing tests.

Take a look at this one:
```

[13:30:56][Test web UI] FAILED TESTS:
[13:30:56][Test web UI]   <SortedTable>
[13:30:56][Test web UI]     ✖ correctly sorts data based on sortSetting
[13:30:56][Test web UI]       HeadlessChrome 79.0.3945 (Linux 0.0.0)
[13:30:56][Test web UI]     AssertionError: first columns match: expected 'c' to equal 'a'
[13:30:56][Test web UI]         at eval (webpack:///./src/views/shared/components/sortedtable/sortedtable.spec.tsx?:141:53)
[13:30:56][Test web UI]         at arrayEach (webpack:///./node_modules/lodash/lodash.js?:47:113)
[13:30:56][Test web UI]         at Function.forEach (webpack:///./node_modules/lodash/lodash.js?:4226:101)
[13:30:56][Test web UI]         at assertMatches (webpack:///./src/views/shared/components/sortedtable/sortedtable.spec.tsx?:139:53)
[13:30:56][Test web UI]         at Context.eval (webpack:///./src/views/shared/components/sortedtable/sortedtable.spec.tsx?:151:5)
[13:30:56][Test web UI] 
```

> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46416#-:-M3RtwMI5PluN92FKcNX:bih8s9q)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/shared/components/sortedtable/index.tsx, line 122 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46416#-M3RxeJ811ejOw716r0W:-M3RxeJ811ejOw716r0X:bj8m4dm) ([raw file](https://github.com/cockroachdb/cockroach/blob/0b846131943bab69fb185cac4b873867ab65fc61/pkg/ui/src/views/shared/components/sortedtable/index.tsx#L122)):*
> ```XML
>     (data: T[], sortSetting: SortSetting, columns: ColumnDescriptor<T>[]): T[] => {
>       if (!sortSetting) {
>         return this.paginatedData();
> ```

Just in terms of reading the code, it's surprising to see pagination being referenced in this method. I would not expect the sorting of data to "know" anything about pagination since pagination gets applied afterwards.

Can we rename `sorted` to `sortedAndPaginated`? That's probably the easiest way to clarify.
___
*[pkg/ui/src/views/statements/statementsPage.tsx, line 119 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46416#-M3S-fTx07lHbnn92JSf:-M3S-fTx07lHbnn92JSg:bjm8911) ([raw file](https://github.com/cockroachdb/cockroach/blob/322f83364bc0807c3147af0631cb7317c584ce43/pkg/ui/src/views/statements/statementsPage.tsx#L119)):*
> ```XML
>   }
> 
>   getStatementsData = () => {
> ```

Really like that this change is centralizing the sorted table pagination! :clap: :tada: 


<!-- Sent from Reviewable.io -->

dhartunian commented:
Release note: "this fixes a bug where sort columns were only being applied per-page instead of for the entire multi-page list of statements"

vladlos committed:
ui: Sort Table test

Added tests for `Sort Table` pagination with sort settings

Resolves: #46173

Release justification: bug fix

Release note (ui): this fixes a bug where sort columns were only being applied per-page instead of for the entire multi-page list of statements

dhartunian reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> thanks for the test!

> Reviewed 1 of 3 files at r1, 1 of 1 files at r2, 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46416#-:-M3lEob_AFbrpmoWKRLm:b-9d8ztr)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @elkmaster)



<!-- Sent from Reviewable.io -->

vladlos committed:
ui: Sortable sort selector

remaned `sorted` createSelector to `sortedAndPaginated`

Resolves: #46173

Release justification: bug fix

Release note (ui): this fixes a bug where sort columns were only being applied per-page instead of for the entire multi-page list of statements

dhartunian commented:
bors r+
